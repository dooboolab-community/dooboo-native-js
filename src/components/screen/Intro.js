// @flow
import React, { Component } from 'react';
import {
  Platform,
  StatusBar,
  StyleSheet,
  TouchableHighlight,
  TouchableOpacity,
  Image,
  ScrollView,
  Text,
  View,
  FlatList,
  InteractionManager,
} from 'react-native';
import { observable } from 'mobx';
import { observer } from 'mobx-react';
import { inject } from 'mobx-react/native';
import { NavigationScreenProp, NavigationStateRoute } from 'react-navigation';

import { ratio, colors } from '../../utils/Styles';
import { IC_MASK } from '../../utils/Icons';
import User from '../../models/User';
import { getString } from '../../../STRINGS';
import Button from '../shared/Button';

const styles: any = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.background,
    flexDirection: 'column',
    alignItems: 'center',
  },
  titleTxt: {
    marginTop: 100,
    color: colors.dusk,
    fontSize: 24,
  },
  txtLogin: {
    fontSize: 14,
    color: 'white',
  },
  imgBtn: {
    width: 24,
    height: 24,
    position: 'absolute',
    left: 16,
  },
  viewUser: {
    marginTop: 60,
    alignItems: 'center',
  },
  txtUser: {
    fontSize: 16,
    color: colors.dusk,
    lineHeight: 48,
  },
  btnBottomWrapper: {
    position: 'absolute',
    bottom: 40,
  },
  btnLogin: {
    backgroundColor: colors.dodgerBlue,
    alignSelf: 'center',
    borderRadius: 4,
    width: 320,
    height: 52,

    alignItems: 'center',
    justifyContent: 'center',
  },
  btnNavigate: {
    backgroundColor: 'white',
    alignSelf: 'center',
    borderRadius: 4,
    width: 320,
    height: 52,

    alignItems: 'center',
    justifyContent: 'center',
  },
});

type Props = {
  store: any;
  navigation: NavigationScreenProp<NavigationStateRoute>;
};
type State = {
  isLoggingIn: boolean;
}

@inject('store') @observer
class Page extends Component<Props, State> {
  timer: any;

  state = {
    isLoggingIn: false,
  };

  componentWillUnmount() {
    if (this.timer) {
      clearTimeout(this.timer);
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.titleTxt}>DOOBOO NATIVE</Text>
        <View style={styles.viewUser}>
          <Text style={styles.txtUser}>{this.props.store.user.displayName}</Text>
          <Text style={styles.txtUser}>{this.props.store.user.age}</Text>
          <Text style={styles.txtUser}>{this.props.store.user.job}</Text>
        </View>
        <View style={styles.btnBottomWrapper}>
          <Button
            isLoading={this.state.isLoggingIn}
            onPress={this.onLogin}
            style={styles.btnLogin}
            textStyle={styles.txtLogin}
            imgLeftSrc={IC_MASK}
            imgLeftStyle={styles.imgBtn}
          >{getString('LOGIN')}</Button>
          <Button
            onPress={() => this.props.navigation.navigate('Temp') }
            style={[
              styles.btnNavigate,
              {
                marginTop: 15,
              },
            ]}
            textStyle={{
              color: colors.dodgerBlue,
            }}
          >Navigate</Button>
        </View>
      </View>
    );
  }

  onLogin = () => {
    this.props.store.user = new User();
    this.setState({ isLoggingIn: true }, () => {
      this.timer = setTimeout(() => {
        this.props.store.user.displayName = 'dooboolab';
        this.props.store.user.age = 30;
        this.props.store.user.job = 'developer';
        this.setState({ isLoggingIn: false });
      }, 1000);
    });
  }
}

export default Page;
