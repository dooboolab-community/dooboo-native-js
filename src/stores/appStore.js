import { observable } from 'mobx';
import { NavigationActions } from 'react-navigation';

import User from '../models/User';

class ObservableListStore {
  @observable rootNavigatorActionHorizontal: boolean = true;
  @observable user: User = new User();
}

const observableListStore = new ObservableListStore();
export default observableListStore;
