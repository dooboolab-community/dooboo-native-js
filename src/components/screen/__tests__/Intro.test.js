import 'react-native';
import * as React from 'react';
import Intro from '../Intro';
import appStore from '../../../stores/appStore';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow, render } from 'enzyme';

describe('Intro', () => {
  // it('renders without crashing', () => {
  //   const rendered = renderer.create(<Intro store={appStore}/>).toJSON();
  //   expect(rendered).toMatchSnapshot();
  //   expect(rendered).toBeTruthy();
  // });

  describe('component test', () => {
    const wrapper = shallow(
      <Intro store={appStore} />,
    );

    it('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
