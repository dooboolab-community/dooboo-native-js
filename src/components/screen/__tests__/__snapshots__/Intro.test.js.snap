// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro component test renders as expected 1`] = `
<Page
  store={
    ObservableListStore {
      "rootNavigatorActionHorizontal": undefined,
      "user": User {
        "age": undefined,
        "displayName": undefined,
        "job": undefined,
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "User@3.keys",
            "observers": Set {},
          },
          "name": "User@3",
          "target": [Circular],
          "values": Map {
            "displayName" => undefined,
            "age" => undefined,
            "job" => undefined,
          },
        },
      },
      Symbol(mobx did run lazy initializers): true,
      Symbol(mobx administration): ObservableObjectAdministration$$1 {
        "defaultEnhancer": [Function],
        "keysAtom": Atom$$1 {
          "diffValue": 0,
          "isBeingObserved": false,
          "isPendingUnobservation": false,
          "lastAccessedBy": 0,
          "lowestObserverState": 2,
          "name": "ObservableListStore@2.keys",
          "observers": Set {},
        },
        "name": "ObservableListStore@2",
        "target": [Circular],
        "values": Map {
          "rootNavigatorActionHorizontal" => undefined,
          "user" => User {
            "age": undefined,
            "displayName": undefined,
            "job": undefined,
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "User@3.keys",
                "observers": Set {},
              },
              "name": "User@3",
              "target": [Circular],
              "values": Map {
                "displayName" => undefined,
                "age" => undefined,
                "job" => undefined,
              },
            },
          },
        },
      },
    }
  }
/>
`;

exports[`Intro renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(65, 77, 107)",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 28,
        "marginTop": 100,
      }
    }
  >
    DOOBOO NATIVE
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 80,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#eee",
          "fontSize": 16,
          "lineHeight": 48,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#eee",
          "fontSize": 16,
          "lineHeight": 48,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#eee",
          "fontSize": 16,
          "lineHeight": 48,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "bottom": 40,
        "position": "absolute",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "white",
            "borderRadius": 4,
            "borderWidth": 2,
            "height": 52,
            "justifyContent": "center",
            "width": 320,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/icons/mask.png",
            }
          }
          style={
            Object {
              "height": 24,
              "left": 16,
              "position": "absolute",
              "width": 24,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
            }
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
